src/deploy.ts

```diff
@@ -0,0 +1,33 @@
+import { DeployOptions, DeployResult } from "hardhat-deploy/dist/types";
+import { WrappedPolyjuiceProvider } from "./provider";
+
+export const patchDeploy = (
+  hre: any,
+  wrappedPolyjuiceProvider: WrappedPolyjuiceProvider
+) => {
+  if (hre.deployments) {
+    const oldDeploy = hre.deployments.deploy;
+    const newDeploy = async (
+      name: string,
+      options: DeployOptions
+    ): Promise<DeployResult> => {
+      if (options.args) {
+        const contractArtifact = await hre.deployments.getExtendedArtifact(
+          name
+        );
+        options.args = await wrappedPolyjuiceProvider.polyjuiceWallet.convertDeployArgs(
+          options.args,
+          contractArtifact.abi,
+          contractArtifact.bytecode
+        );
+      }
+
+      return oldDeploy(name, options);
+    };
+    hre.deployments.deploy = newDeploy;
+  } else {
```

Comment:
it's more readable when you revert it.
you have:

```
if (importantCondition) {
  doStuff()
} else {
  reportError()
}
```

It's definitely recommended to always to this as:
```
if (!importantCondition) {
   reportError()
   return
}
doStuff()
```